packages/remark-math/inline.js

```diff
@@ -6,35 +6,46 @@ const ESCAPED_INLINE_MATH = /^\\\$/ // starts with \$
 
 const NODE_TYPES = {
   TEXT: 'text',
-  MATH_INLINE: 'inlineMath'
+  INLINE_MATH: 'inlineMath'
+}
+
+const includeNonDollarsAndEscaped = ['\\$', /[^$]/]
+
+function escapeRegExp (str) {
+  return str.replace(/[.*+?^${}()|[\]\\]/g, '\\$&')
```

Comment:
```suggestion
  return str.replace(/[.*+\-?^${}()|[\]\\]/g, '\\$&');
```

Tweaking this to match the MDN docs instead of the StackOverflow answer, because it seems slightly improved since it was copied to SO:

https://wiki.developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions$compare?locale=en-US&to=1607265&from=1605149